HW3.py

```diff
@@ -0,0 +1,116 @@
+# Created on December 26, 2019
+
+
+import sys
+import re
+import operator
+
+NO_GUESSES = "I don't know any word of a given length..."
+ONE_MORE_TRY = 'Would you like to play one more times? (y/n): '
+YES = 'Y'
+BYE = 'Bye, see you next time! :)'
+WELCOME ='Welcome to a Guessing Word Game!'
+THINK_A_WORD = "Think of a word and enter it's length: "
+POSITIONS_OF_AVAILABLE_LETTERS ='\n If letter(s): "{}" is available in the hidden word, \n please enter comma separated position(s) counting from 0 like 0,2, \n or just press Enter if the letter is absent: '
+EMPTY_STRING = ''
+COMMA = ','
+DOT = '.'
+GUESSED_WORD = 'I want to tell wholw word: "{}"'
+CANT_GUESS = "You are che champions! Can't guess the word matching specified letters"
+NO_TRIES_LEFT = "No more word, so you're probably cheating, stop doing it!"
+PATH_TO_DATABASE = 'C:\Python\words.txt'
+
+def searchWords(textFile):
+    text = re.findall (r'\w+', textFile.read().lower())
+    return text
+
+def splitWord(word):
+    return [char for char in word]
+
+def sumFreq(d):
+    sum_freq = 0
+    for key in d:
+        sum_freq += d[key]
+    return sum_freq
+
+def searchUniqueWords(inputList,givenWordLength):
+    listSet = set(inputList)
+    unique_words_list = sorted(list(listSet))
+    possibleWordsMatchedReg = [el.lower() for el in unique_words_list if len(el) == givenWordLength]
+    if possibleWordsMatchedReg==[]:
+        print (NO_GUESSES)
+        newGame()
+    else:
+        return possibleWordsMatchedReg
+
+def getCharFrequency(wordSubset):
+    wordsString = EMPTY_STRING.join(wordSubset)
+    result = {}
+    freq = {}
+    sumFreqChars =0
+    for char in wordsString:
+        try:
+            result[char] +=1  # If char already in result dictionary, increase its count
+            sumFreqChars +=1
+        except KeyError:
+            result[char] = 1  # If char not in result dict yet, start counting it
+            sumFreqChars +=1
+    for char in result:
+        freq[char]=round(result[char]*100/sumFreqChars,2)
+
+    result = sorted(freq.items(), key=operator.itemgetter(1), reverse=True)
+    return result
+
+def newGame():
```

Comment:
It doesn't clear the difference between newGame() and start_game() functions.
It'd better to rename it to "ask_new_game()"